src/main/scala/util/RecordMap.scala

```diff
@@ -0,0 +1,27 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.util
+
+import Chisel._
+import chisel3.Record
+import scala.collection.immutable.ListMap
+
+final case class RecordMap[T <: Data](eltMap: ListMap[String, T])
+    extends Record with collection.Seq[T] {
```

Comment:
I don't think we should extend `Seq`, I think the only thing we should extend is `Record`. `Seq` is less problematic than `Map`, but there is the issue that it may be surprising.

Reply:
so the common pattern is to want to zip these things together. Can I still do that if it doesnt' extend Seq?

Reply:
though there is an argument to be made that zipping them up sequentially (vs pairing them by name) is exactly the buggy case we want to avoid... thoughts @hcook ?

Reply:
`.elements` is of type `ListMap` which implements `Iterable` so you can always `zip` those, eg.
```scala
whatever.zip(myRecordMap.elements)
```

Reply:
To be clear... `elements` is not the REAL elements, right? Given your suggestion below, elements returns some sort of cloned type version thing, wheras eltMap.values gives... the real data? 

I wrote a `val data = eltMap.values` function separately from `elements` because it seemed more correct/different for the use case when you actually want to connect to the bundle, vs clone its type. But I'm having difficulty getting this right.